### ISSUE TITLE
Witnessing a pool registration/modification with HW wallet fails

### ISSUE THREAD

**Issue by amani1104** [cardano-community/guild-operators#692]
**Describe the bug**
When witnessing a pool registration with multiple owners of which at least one is a HW wallet (in this case a Ledger Nano S), the witness operation after signing with the HW wallet fails. The error message is the following: 

```
Error: Tx serialization mismatch
jq: error: syntax error, unexpected '}' (Unix shell quoting issues?) at <top-level>, line 1:
.witness += [{ name: "Owner #3 'letzbake' stake signing key", witnessBody:  }]                                                                            
jq: 1 compile error
```

**To Reproduce**
1. Sign the witness operation with HW wallet

**Expected behavior**
The operation does not fail.

**Version:**
 - OS: Ubuntu 20.04 LTS
 - Product version: CNTools v7.0.0 alpha
 - Cardano Node version: 1.24.2
 - Network you're connecting to: Mainnet

**Comment by Scitz0** [comment]
Pool registration is not supported with current firmware/app in ledger yet. Release cooking soon. But I will look it over to provide a better error message.

**Comment by Scitz0** [comment]
Actually, I just noticed ledger released 2.1.0. I will verify that pool reg work with this version before pushing 7.0.0 to master.

**Comment by amani1104** [comment]
Thanks @Scitz0. I assumed that 2.1.0 is the new version.

**Comment by Scitz0** [comment]
> Thanks @Scitz0. I assumed that 2.1.0 is the new version.

It is, if this is the case... then I have some bug hunting to do :)
Testing now myself, haven't had a chance to play with 2.1.0 yet.

**Comment by Scitz0** [comment]
Hopefully solved by #694

**Comment by amani1104** [comment]
I cannot test it, unfortunately, as line 79 in cntools.library you modified:

```
78 distanceToBottom() {
79   [[ $(( $(tput lines) - $(get_cursor_row) )) -lt 5 ]] && printf '\n\n\n\n\n\033[5A' >&6
80 }
```

causes the following error, when trying to list wallets: 

```
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 >> WALLET >> LIST
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

/opt/cardano/cnode/scripts/cntools.library: line 79: 34 - /opt/cardano/cnode/priv/wallet/hw_rewards/opt/cardano/cnode/priv/wallet/letzbake/opt/cardano/cnode/priv/wallet/letzbake_rewards/opt/cardano/cnode/priv/wallet/vote/opt/cardano/cnode/priv/wallet/witness : syntax error: operand expected (error token is "/opt/cardano/cnode/priv/wallet/hw_rewards/opt/cardano/cnode/priv/wallet/letzbake/opt/cardano/cnode/priv/wallet/letzbake_rewards/opt/cardano/cnode/priv/wallet/vote/opt/cardano/cnode/priv/wallet/witness ")
```

**Comment by amani1104** [comment]
I reverted the modifications you made concerning "distanceToBottom" to test the changes. I now receive the following error when signing the witness operation on the HW device (Ledger Nano S): 

`Shelley command failed: transaction sign-witness  Error: /tmp/cntools/tx.witness_d4jEL54Xzc: TextEnvelope type error:  Expected one of: TxWitnessByron, TxWitnessShelley, TxWitness AllegraEra, TxWitness MaryEra Actual: `

**Comment by Scitz0** [comment]
1) I can confirm List issue, makes it easier to fix :)

2) Can you please provide the full content of the offline file generated?

**Comment by Scitz0** [comment]
Solved in official CNTools 7.0.0 merged to master
